bevy_lint/src/lints/borrowed_reborrowable.rs

```diff
@@ -0,0 +1,233 @@
+//! Checks for function parameters that take a mutable reference to a
+//! re-borrwable type.
+//!
+//! # Motivation
+//!
+//! Mutable references to re-borrwable types can almost always be readily
+//! converted back to an owned instance of itself, albeit with an appropriately
+//! shorter lifetime.
+//!
+//! The only time this isn't true is when the function returns referenced data
+//! that is bound to the mutable reference of the re-borrowable type.
+//!
+//! # Example
+//!
+//! ```
+//! # use bevy::prelude::*;
+//! fn system(mut commands: Commands) {
+//!   helper_function(&mut commands);
+//! }
+//!
+//! fn helper_function(commands: &mut Commands) {
+//!   // ...
+//! }
+//! ```
+//!
+//! Use instead:
+//!
+//! ```
+//! # use bevy::prelude::*;
+//! fn system(mut commands: Commands) {
+//!   helper_function(commands.reborrow());
+//! }
+//!
+//! fn helper_function(mut commands: Commands) {
+//!   // ...
+//! }
+//! ```
+//!
+//! # Known Issues
+//!
+//! This lint does not currently support closures.
```

Comment:
Could you move this to be right below the Motivation section, just like [`missing_reflect`](https://github.com/TheBevyFlock/bevy_cli/blob/2480d608b4cb527a01c463b89b620e626f7108c7/bevy_lint/src/lints/missing_reflect.rs#L10)?